dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java

```diff
@@ -572,4 +573,83 @@ public void testLoadCategories_GivenContentWithCats_ShouldLoadESMapWithListOfCat
+        final ContentType contentType = new ContentTypeDataGen()
+                .name(contentTypeName)
+                .velocityVarName(contentTypeName)
+                .fields(fields)
+                .nextPersisted();
+
+        new ContentletDataGen(contentType.id()).setProperty(myKeyValueField,
+                "{\"key\":\"key1\", value:\"val\" }").nextPersisted();
+        final String queryString =  String.format("+%s.%s.key:%s*",contentTypeName, myKeyValueField, "val");
+        Logger.info(ESMappingAPITest.class, () -> String.format(" Query: %s ",queryString));
+        final String wrappedQuery = String.format("{"
+                + "query: {"
+                + "   query_string: {"
+                + "        query: \"%s\""
+                + "     }"
+                + "  }"
+                + "}", queryString);
+
+        final ESSearchResults searchResults = contentletAPI.esSearch(wrappedQuery, false,  user, false);
+        Assert.assertFalse(searchResults.isEmpty());
+        for (Object searchResult : searchResults) {
+           final Contentlet contentlet = (Contentlet) searchResult;
+            final String json = (String)contentlet.getMap().get("myKeyValueField");
+            Assert.assertNotNull(json);
+            final Map<String, Object> map = KeyValueFieldUtil.JSONValueToHashMap(json);
+            Assert.assertTrue(map.containsKey("key"));
+        }
+    }
+
+    /**
+     * General purpose is testing that Metadata fields are indexed correctly can be used as query parameters.
+     * Given scenario: Known that saving a file assets generates metadata into the index. We create a file asset.
+     * Expected Results: Once the file is saved we use an ES query to verify that queries by metadata are functional.
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void Test_Create_FileAsset_Query_by_MetaData_Expect_Success()
+            throws DotDataException, DotSecurityException {
+
+        final File binary = new File(ESMappingAPITest.class.getClassLoader().getResource("images/test.jpg").getFile());
+        final Host site = new SiteDataGen().nextPersisted();
+        final FileAssetDataGen fileAssetDataGen = new FileAssetDataGen(site, binary);
+        fileAssetDataGen.nextPersisted();
+
+        final String queryString =  String.format("+fileasset.filename:%s +metadata.contenttype:image* +metadata.filesize:%d ",binary.getName(), binary.length());
+        Logger.info(ESMappingAPITest.class, () -> String.format(" Query: %s ",queryString));
+        final String wrappedQuery = String.format("{"
+                + "query: {"
+                + "   query_string: {"
+                + "        query: \"%s\""
+                + "     }"
+                + "  }"
+                + "}", queryString);
+
+        final ESSearchResults searchResults = contentletAPI.esSearch(wrappedQuery, false,  user, false);
+        Assert.assertFalse(searchResults.isEmpty());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertFalse' due to existing static import 'org.junit.Assert.assertFalse'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5533307)